src/main/java/seedu/gamebook/model/gameentry/GameEntry.java

```diff
@@ -158,8 +158,6 @@ public boolean isSameGameEntry(GameEntry otherGameEntry) {
         }
         return equals(otherGameEntry)
                 || (gameType.equals(otherGameEntry.gameType)
-                        && date.getIsTimeIndicated()
-                        && otherGameEntry.date.getIsTimeIndicated()
```

Comment:
Perhaps an assertion can be added prior to the return statement to check that DatePlayed#equals() already takes care of checking whether date is indicated to improve defensiveness of code.